frontend/server/src/RabbitMQConnection.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace OmegaUp;
+
+class RabbitMQConnection {
+    /**
+     * The singleton instance of this class.
+     * @var null|RabbitMQConnection
+     */
+    private static $_instance = null;
+
+    /**
+     * @psalm-readonly
+     */
+    public \PhpAmqpLib\Connection\AMQPStreamConnection $connection;
+
+    /**
+     * Returns the singleton instance of this class. It also registers a
+     * shutdown function to close the connection upon script
+     * termination.
+     */
+    public static function getInstance(): RabbitMQConnection {
+        if (is_null(self::$_instance)) {
+            self::$_instance = new RabbitMQConnection();
+            register_shutdown_function(function () {
+                if (is_null(self::$_instance)) {
+                    return;
+                }
+                self::$_instance->connection->close();
+                self::$_instance = null;
+            });
+        }
+        return self::$_instance;
+    }
+
+    private function __construct() {
+        $this->connection = new \PhpAmqpLib\Connection\AMQPStreamConnection(
+            OMEGAUP_RABBITMQ_HOST,
+            OMEGAUP_RABBITMQ_PORT,
+            OMEGAUP_RABBITMQ_USERNAME,
+            OMEGAUP_RABBITMQ_PASSWORD,
+        );
+    }
+
+    /**
+     * @return mixed
+     */
+    public function channel() {
+        if (!is_null(self::$_instance)) {
+            return self::$_instance->connection->channel();
+        }
```

Comment:
```suggestion
        return $this->connection->channel();
```

aquí es conveniente evitar la posibilidad de que se regrese nulo. como la función no es estática, tenemos la garantía de que `$this->connection` no es nulo nunca.

Reply:
Hola, lo puse asi como me lo sugieres y al momento de correr el validador de tipos es cuando me marca ese error de posible valor nulo

Reply:
huh, puedes pegar el error exacto para no parafrasear?

Reply:
```
ERROR: PossiblyNullReference - frontend/server/src/RabbitMQConnection.php:49:46 - Cannot call method channel on possibly null value (see https://psalm.dev/083)
        return self::$_instance->connection->channel();
```

Reply:
hmm, eso no corresponde con la sugerencia. puedes usar `$this->connection` en vez de `self::$_instance->connection`?

Reply:
:o, genial, asi si funciono